src/treeview/treeview.component.ts

```diff
@@ -170,4 +170,13 @@ export class TreeViewComponent implements AfterViewInit, OnInit, OnDestroy {
 	public isProjected() {
 		return this.treeViewService.contentProjected;
 	}
+
+	private copyNode(node: Node): Node {
+		// making a recursive shallow copy to avoid performance issues when deeply cloning templateRefs if defined in the node
+		const copiedNode = Object.assign({}, node)
+		if (node.children) {
+		  copiedNode.children = node.children.map(child => this.copyNode(child))
+		}
+		return copiedNode
+	}
```

Comment:
```suggestion

	private copyNode(node: Node): Node {
		// making a recursive shallow copy to avoid performance issues when deeply cloning templateRefs if defined in the node
		const copiedNode = Object.assign({}, node)
		if (node.children) {
		  copiedNode.children = node.children.map(child => this.copyNode(child))
		}
		return copiedNode
	}
```
```suggestion

	private copyNode(node: Node): Node {
		// making a recursive shallow copy to avoid performance issues when deeply cloning templateRefs if defined in the node
		const copiedNode = Object.assign({}, node);
		if (node.children) {
		  copiedNode.children = node.children.map(child => this.copyNode(child));
		}
		return copiedNode;
	}
```